internal/resources/schemas/lb_profile_schemas.go

```diff
@@ -0,0 +1,426 @@
+package schemas
+
+import (
+	"github.com/HewlettPackard/hpegl-vmaas-terraform-resources/internal/resources/validations"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func HttpProfileSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Optional:    true,
+		Description: "HTTP Profile configuration",
+		MaxItems:    1,
+		ExactlyOneOf: []string{"http_profile", "tcp_profile", "udp_profile",
+			"cookie_profile", "sourceip_profile", "generic_profile",
+			"client_profile", "server_profile"},
+		ConflictsWith: []string{"tcp_profile", "udp_profile",
+			"cookie_profile", "sourceip_profile", "generic_profile",
+			"client_profile", "server_profile"},
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"service_type": {
+					Type:             schema.TypeString,
+					ValidateDiagFunc: validations.StringInSlice([]string{"LBHttpProfile", "LBFastTcpProfile", "LBFastUdpProfile", "LBClientSslProfile", "LBServerSslProfile", "LBCookiePersistenceProfile", "LBGenericPersistenceProfile", "LBSourceIpPersistenceProfile"}, false),
+					Required:         true,
+					Description:      "Network Loadbalancer Supported values are `LBHttpProfile`,`LBFastTcpProfile`, `LBFastUdpProfile`, `LBClientSslProfile`,`LBServerSslProfile`, `LBCookiePersistenceProfile`,`LBGenericPersistenceProfile`,`LBSourceIpPersistenceProfile`",
+				},
+				"http_idle_timeout": {
+					Type:     schema.TypeInt,
+					Required: true,
+					//Default:          15,
+					ValidateDiagFunc: validations.IntBetween(1, 5400),
+					Description:      "http_idle_timeout for Network Load balancer Profile",
+				},
+				"request_header_size": {
+					Type:     schema.TypeInt,
+					Required: true,
+					//Default:          1024,
+					ValidateDiagFunc: validations.IntBetween(1, 65536),
+					Description:      "request_header_size for Network Load balancer Profile",
+				},
+				"response_header_size": {
+					Type:             schema.TypeInt,
+					Required:         true,
+					ValidateDiagFunc: validations.IntBetween(1, 65536),
+					//Default:          4096,
+					Description: "response_header_size for Network Load balancer Profile",
+				},
+				"redirection": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "redirection for Network Load balancer Profile",
+				},
+				"x_forwarded_for": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Default:     "INSERT",
```

Comment:
what are the other values can it take other than INSERT, can it be a enum list ?

Reply:
yes added that